yapf/yapflib/file_resources.py

```diff
@@ -98,7 +98,7 @@ def _FindPythonFiles(filenames, recursive):
 
 def IsPythonFile(filename):
   """Return True if filename is a Python file."""
-  if os.path.splitext(filename)[1] == '.py':
+  if os.path.splitext(filename)[1] in ['.py', '.gyp', '.gypi']:
```

Comment:
@gwelymernans may I suggest a different approach altogether? Instead of checking the file type, we can use the `ast` syntax tree module to check if the content is Python compatible. The [parser](https://docs.python.org/3/library/ast.html#ast.parse) raises a `SyntaxError`  if the code isn't Pythonic.

Reply:
You probably want to add `'.pyi'` to the list. (It'll also work with the `ast` check)

One problem with the `ast` check is that a Python file with a syntax error will just be skipped. Currently, it produces a rather unattractive crash. Whether this is a bug or a feature depends on the use case -- if an editor calls yapf (e.g., using emacs' flymake), it could be useful to mark the syntax error rather than silently do nothing.

Reply:
Use ['.py', '.gyp', '.gypi'] instead of ['.py', '.gyp', '.gypi'] because:
1. tuples are smaller and more efficient than lists
2. a tuple signals to the reader that the contents are fixed and can not be modified at runtime.

I am not a fan of the ast approach for performance reasons and because it would not alert users of malformed Python files.

Reply:
I agree with @cclauss. Parsing the file more than once will make things slower than need be. We already have performance problems, because the base algorithm is O(N^2). It could also lead to yapf trying to modify a file that isn't meant to be formatted (i.e., it's not a Python file, but the `ast` parser parses it without error).

Another issue is that if we hardcode the file extensions, then we'll have to change the list for any future additions, which is kinda annoying and makes me think that there should be a Better Way(tm). I suggested adding it as an option to .yapf.style, or command line parameter, etc., for that reason. I'm not saying that we need to do it that way, but I would like a more flexible option.